receiver/kafkametricsreceiver/receiver.go

```diff
@@ -42,12 +43,21 @@ var (
 	}
 )
 
+func logDeprecation(logger *zap.Logger) {
+	logger.Warn("kafka.brokers attribute is deprecated and will be removed in a future release. Use kafka.brokers.count instead.")
+}
+
 var newMetricsReceiver = func(
 	ctx context.Context,
 	config Config,
 	params receiver.CreateSettings,
 	consumer consumer.Metrics,
 ) (receiver.Metrics, error) {
+
+	if config.Metrics.KafkaBrokers.Enabled {
+		logDeprecation(params.Logger)
+	}
```

Comment:
You don't need to do this manually `warnings.if_enabled` metadata generates code for that https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/14167/files#diff-fef666704b86e8e7ac006eb203d4ba539eb4295bf714e77890a2a1e6313071b4R1282

Reply:
thanks. Sorry the long standing pr and the delay. I have pushed up these changes.